plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -306,57 +339,50 @@ def save(self):
         Update the DB model to store the downloaded file.
         Then, save in the DB.
         """
-        artifact = None
-
-        if not self.fetched:
-            if self._path:  # downloaded
-                try:
-                    with transaction.atomic():
-                        self.model = Artifact(
-                            file=File(open(self._path, mode='rb')),
-                            **self.monitor.facts())
-                        self.model.save()
-                except IntegrityError:
-                    q = self.artifact_q()
-                    self.model = Artifact.objects.get(q)
-                finally:
-                    artifact = self.model
-        else:
-            artifact = self.model
+        if self._path and (not self._stored_model):  # downloaded
+            try:
+                with transaction.atomic():
+                    self._stored_model = Artifact(
+                        file=File(open(self._path, mode='rb')),
+                        **self._monitor.facts())
+                    self._stored_model.save()
+            except IntegrityError:
+                q = self.artifact_q()
+                self._stored_model = Artifact.objects.get(q)
 
         try:
             with transaction.atomic():
                 content_artifact = ContentArtifact(
                     relative_path=self.relative_path,
-                    content=self.content.model,
-                    artifact=artifact)
+                    content=self.content.stored_model,
+                    artifact=self._stored_model)
                 content_artifact.save()
         except IntegrityError:
             content_artifact = ContentArtifact.objects.get(
                 relative_path=self.relative_path,
-                content=self.content.model)
-            if self.fetched:
-                content_artifact.artifact = artifact
+                content=self.content.stored_model)
+            if self._stored_model:
+                content_artifact.artifact = self._stored_model
                 content_artifact.save()
 
-        digests = {f: getattr(self.model, f) for f in Artifact.DIGEST_FIELDS}
+        digests = {f: getattr(self._model, f) for f in Artifact.DIGEST_FIELDS}
```

Comment:
This fixes issue because you now take data from "pending" model here and not from the stored one. And before this PR there was no such separation and you updated the only model you had. Correct?

Reply:
Heh, now I see your comment to this PR about separation between pending and stored models :)

Reply:
Exactly.